---
title: "Project 01.01 Report"
author: "Annika Hein, Amanda Janzen, Nicolai Laufer, Niklas Schwan"
date: "25/06/2019"
output: html_document
---
# Introduction

In this project we tried to identify genetic interactions and their effects on several Glioblastoma multiforme (GBM) cell lines. 

GBM is a fast growing, malignant brain tumor classified as a grade IV cancer by the World Health Organisation (WHO). It represents about 15% of all brain tumors and most commonly develops in adults aged 45-70. We chose GBM because it falls in the category of cancers with poor diagnosis and also has no effective long-term treatments, which leads to a survival rate of less than 15 months, according to the TCGA (The Cancer Genome Atlas Research Network).

The identification of genomic interactions and their effect on cancer cell survival is very important for the search after a better treatment. Its concept is tied to the fact that cancer is a product of accumulated genomic defects (mutations, copy number changes, etc.). There are a handful of genes that drive the process of oncogenesis, these are called *driver mutations* (DM), all other mutations that don't really have an impact on the oncogenisis are called *passenger mutations*.

The driver mutations are normally highly essential genes for all cells in the human body, making them a bad target for anti-cancer drugs, for they would shut down all cells, not just cancer cells. The alternative to this targeting problem is to find genes that are linked with the driving mutations in the cancer cells, but not in the normal cells. Ideally, the targeting of these so called *second site targets* (SST) would cause the cancer cells to shut down and stop proliferating, while the normal cells would not be subjected to any changes.

Our aim in this project was to find the driver mutations in GBM throughout all cell lines in our data set, compare them to the literature and investigate if there is a dependency between them and any second site targets. In the end the aim was to find a model where we could possibly predict a second site mutation based on the existence of a driving mutation.

# Data cleanup

* Once we downloaded the data, we first removed all data that didn't have to do with Glioblastoma.

```
brain.anno = allDepMapData$annotation[allDepMapData$annotation$Subtype.Disease == "Glioblastoma", ]
```
* Then we made the cell line identifiers be the rownames in the annotation.

```
rownames(brain.anno) = brain.anno$DepMap_ID
brain.anno = brain.anno[, -1]
```
* The next step was to remove all cell lines that were irrelevant to our project from all the other matrices (expression, copy number, relevant mutations, ceres and probability)

** "operators" package needed for this step, because of "%!in%"

```
cell.lines = dput(rownames(brain.anno))
exp.clean = allDepMapData$expression[, -which(colnames(allDepMapData$expression) %!in% cell.lines)]
copy.clean = allDepMapData$copynumber[, -which(colnames(allDepMapData$copynumber) %!in% cell.lines)]
relevant.mutations = subset(allDepMapData$mutation, names(allDepMapData$mutation) %in% cell.lines)
ceres.clean = allDepMapData$kd.ceres[, -which(colnames(allDepMapData$kd.ceres) %!in% cell.lines)]
prob.clean = allDepMapData$kd.prob[, -which(colnames(allDepMapData$kd.prob) %!in% cell.lines)]
```

* After we separated all glioblastoma cell lines, we removed the NA and all values out of range. The data contained a lot of 0 and negative values, especially the copy number data, where x>=0.

```
sum(is.na(ceres.clean))
sum(is.na(prob.clean))
sum(is.na(copy.clean))
sum(is.na(exp.clean))
sum(is.na(relevant.mutations))
copy.clean = copy.clean[-which(apply(copy.clean, 1, function(x) {sum(is.na(x))}) > 0), ]
sum(is.na(copy.clean))
```
The function sum() was used to show how many NA values were in each matrix.

ceres.clean -> 0
prob.clean -> 0 
copy.clean -> 1780
exp.clean -> 0
relevant.mutations -> 0, but looking directly in the matrix we can see NA values.

```
copy.clean = copy.clean[-which(apply(copy.clean, 1, function(x) {sum(is.na(x))}) > 0), ]
sum(is.na(copy.clean))
```

copy.clean -> 0

* The next cleanup target was probability matrix. It should only contain values between 0 and 1, which is the case.

```
sum(prob.clean < 0) //0
sum(prob.clean > 1) //0
``` 

* Checking for values = 0

```
sum(ceres.clean == 0) 
sum(prob.clean == 0) 
sum(copy.clean == 0) 
sum(exp.clean == 0)
```

ceres.clean -> 28
prob.clean -> 80
copy.clean -> 0
exp.clean -> 644705

* Scan for outliers via boxplots and threshold or remove them if necessary.

Two values were identified an outliers, then thresholded and removed them.

```
boxplot(ceres.clean) //two values identified as outliers
ceres.transf[ceres.clean > 3] = 3 //thresholding
ceres.transf = ceres.clean[-which(ceres.clean > 3), ] //removing
```

* Organising all matrices for further work. 

We compared the gene data availability between all data sets to get the same dimensions and also made all data nominal in the annotation.

```
dim(copy.clean) == dim(exp.clean)
gene.data.co = c(rownames(copy.clean))
gene.data.ex = c(rownames(exp.clean))
exp.clean = exp.clean[-which(rownames(exp.clean) %!in% gene.data.co),]
copy.clean = copy.clean[-which(rownames(copy.clean) %!in% gene.data.ex),]
genes.clean = rownames(exp.clean)
ceres.clean = ceres.clean[-which(rownames(ceres.clean) %!in% genes.clean),]
prob.clean = prob.clean[-which(rownames(prob.clean) %!in% genes.clean),]
genes.clean = rownames(ceres.clean)
exp.clean = exp.clean[-which(rownames(exp.clean) %!in% genes.clean),]
copy.clean = copy.clean[-which(rownames(copy.clean) %!in% genes.clean),]
rm(gene.data.co)
rm(gene.data.ex)

brain.anno$CCLE_Name = factor(brain.anno$CCLE_Name)
brain.anno$Aliases = factor(brain.anno$Aliases)
brain.anno$Primary.Disease = factor(brain.anno$Primary.Disease)
brain.anno$Subtype.Disease = factor(brain.anno$Subtype.Disease)
brain.anno$Subtype.Gender = factor(brain.anno$Gender)
brain.anno$Subtype.Source = factor(brain.anno$Source)
```
dimensions: 16970x28

We also put the gene names and cell lines in alphabetical order

```
ceres.clean = ceres.clean[order(rownames(ceres.clean)), order(colnames(ceres.clean))]
prob.clean = prob.clean[order(rownames(prob.clean)), order(colnames(prob.clean))]
exp.clean = exp.clean[order(rownames(exp.clean)), order(colnames(exp.clean))]
copy.clean = copy.clean[order(rownames(copy.clean)), order(colnames(copy.clean))]
```

* In order to find all most commonly mutated genes among all relevant cell lines, we made a list with all commonly mutated genes and also combined all mutation matrices into one.

```
#common.genes = as.data.frame(table(c(relevant.mutations$`ACH-000036`$Hugo_Symbol,relevant.mutations$`ACH-000040`$Hugo_Symbol, relevant.mutations$`ACH-000075`$Hugo_Symbol, relevant.mutations$`ACH-000098`$Hugo_Symbol, relevant.mutations$`ACH-000208`$Hugo_Symbol, relevant.mutations$`ACH-000215`$Hugo_Symbol, relevant.mutations$`ACH-000137`$Hugo_Symbol, relevant.mutations$`ACH-000152`$Hugo_Symbol, relevant.mutations$`ACH-000591`$Hugo_Symbol, relevant.mutations$`ACH-000673`$Hugo_Symbol, relevant.mutations$`ACH-000231`$Hugo_Symbol, relevant.mutations$`ACH-000244`$Hugo_Symbol, relevant.mutations$`ACH-000760`$Hugo_Symbol, relevant.mutations$`ACH-000368`$Hugo_Symbol, relevant.mutations$`ACH-000376`$Hugo_Symbol, relevant.mutations$`ACH-000445`$Hugo_Symbol, relevant.mutations$`ACH-000464`$Hugo_Symbol, relevant.mutations$`ACH-000469`$Hugo_Symbol, relevant.mutations$`ACH-000479`$Hugo_Symbol, relevant.mutations$`ACH-000570`$Hugo_Symbol, relevant.mutations$`ACH-000571`$Hugo_Symbol, relevant.mutations$`ACH-000623`$Hugo_Symbol, relevant.mutations$`ACH-000631`$Hugo_Symbol, relevant.mutations$`ACH-000738`$Hugo_Symbol, relevant.mutations$`ACH-000756`$Hugo_Symbol, relevant.mutations$`ACH-000819`$Hugo_Symbol, relevant.mutations$`ACH-000128`$Hugo_Symbol, relevant.mutations$`ACH-000887`$Hugo_Symbol)))#

relevant.mutations.combi = do.call(rbind, lapply(which(names(allDepMapData$mutation) %in% cell.lines), function(a) allDepMapData$mutation[[a]]))
common.genes = as.matrix(table(c(relevant.mutations.combi$Hugo_Symbol)))

summary(common.genes)
```

Commonly mutated genes: 
 - MT-ND5 (mitochondrially encoded NADH: ubiquinone oxireductase core subunit 5): 32
 - TTN (titin, large abundant protein of striated muscle): 26
 - MUC16 (mucin): 23
 - MT-CYB (mitochondrially encoded cytochrome b): 18
 - PTEN (phosphatase and tensin homolog): 17
 - other: 10096
 
```
rownames(common.genes) = common.genes$Var1
common.genes$Var1 = NULL
barplot(common.genes, beside = T, names.arg = rownames(common.genes), las = 2)
common.genes.c = subset(common.genes, common.genes$Freq >11)
common.genes.c = common.genes.c[c(6, 11, 10, 7, 4, 9, 3, 2, 1, 8),]
barplot_commongenes <- barplot(common.genes.c, beside = T, names.arg = rownames(common.genes.c), ylab = "Frequency", main = "Most common gene mutations", las = 2)
```

* Combine the mutation matrices of the GBM cell lines, extract the relevant columns "Hugo_Symbol", "IsDeleterious" and "DepMap_ID"
Package dplyr needed for this step

```
mutations.all = rbind(relevant.mutations$`ACH-000036`,relevant.mutations$`ACH-000040`, relevant.mutations$`ACH-000075`, relevant.mutations$`ACH-000098`, relevant.mutations$`ACH-000208`, relevant.mutations$`ACH-000215`, relevant.mutations$`ACH-000137`, relevant.mutations$`ACH-000152`, relevant.mutations$`ACH-000591`, relevant.mutations$`ACH-000673`, relevant.mutations$`ACH-000231`, relevant.mutations$`ACH-000244`, relevant.mutations$`ACH-000760`, relevant.mutations$`ACH-000368`, relevant.mutations$`ACH-000376`, relevant.mutations$`ACH-000445`, relevant.mutations$`ACH-000464`, relevant.mutations$`ACH-000469`, relevant.mutations$`ACH-000479`, relevant.mutations$`ACH-000570`, relevant.mutations$`ACH-000571`, relevant.mutations$`ACH-000623`, relevant.mutations$`ACH-000631`, relevant.mutations$`ACH-000738`, relevant.mutations$`ACH-000756`, relevant.mutations$`ACH-000819`, relevant.mutations$`ACH-000128`, relevant.mutations$`ACH-000887`$Hugo_Symbol)

mutations.all= select(mutations.all, c(2,21, 36))
```

* After that, we determined the GBM cells lines, which contained our four most prominent driver mutations (DM). We then extracted the cell lines which contain one specific mutation (MT-ND5, MUC16, TP53 and TTN) and determined the mean expression, copy number (CN), CERES score and probability of all genes over these cell lines.

For MT-ND5:
```
list.mtnd5 = unique(subset(mutations.all, mutations.all$Hugo_Symbol == "MT-ND5"))
cells.mtnd5 = c(list.mtnd5$DepMap_ID)
mtnd5.exp.mean = as.matrix(c(rowMeans(exp.mtnd5)))
mtnd5.copy.mean = as.matrix(c(rowMeans(copy.mtnd5)))
mtnd5.ceres.mean = as.matrix(c(rowMeans(ceres.mtnd5)))
mtnd5.prob.mean = as.matrix(c(rowMeans(prob.mtnd5)))
```

For MUC16:
```
list.muc16 = unique(subset(mutations.all, mutations.all$Hugo_Symbol == "MUC16"))
cells.muc16 = c(list.muc16$DepMap_ID)
muc16.exp.mean = as.matrix(c(rowMeans(exp.muc16)))
muc16.copy.mean = as.matrix(c(rowMeans(copy.muc16)))
muc16.ceres.mean = as.matrix(c(rowMeans(ceres.muc16)))
muc16.prob.mean = as.matrix(c(rowMeans(prob.muc16)))
```
For TP53:
```
list.tp53 = unique(subset(mutations.all, mutations.all$Hugo_Symbol == "TP53"))
cells.tp53 = c(list.tp53$DepMap_ID)
tp53.exp.mean = as.matrix(c(rowMeans(exp.tp53)))
tp53.copy.mean = as.matrix(c(rowMeans(copy.tp53)))
tp53.ceres.mean = as.matrix(c(rowMeans(ceres.tp53)))
tp53.prob.mean = as.matrix(c(rowMeans(prob.tp53)))
```

For TTN:
```
list.ttn = unique(subset(mutations.all, mutations.all$Hugo_Symbol == "TTN"))
cells.ttn = c(list.ttn$DepMap_ID)
ttn.exp.mean = as.matrix(c(rowMeans(exp.ttn)))
ttn.copy.mean = as.matrix(c(rowMeans(copy.ttn)))
ttn.ceres.mean = as.matrix(c(rowMeans(ceres.ttn)))
ttn.prob.mean = as.matrix(c(rowMeans(prob.ttn)))
```

# Data visualization

* We did boxplots for visualization of whole expression and CN matrices as well as MT-ND5 expression and CN. Also heatmaps for whole mutation, CN, expression, CERES and probability matrices (with and without 0 values).
Packages: pheatmap, grid

Expression matrix (only a few overexpressed genes, a lot of deleted genes) :
```
setHook("grid.newpage", function() pushViewport(viewport(x=1,y=1,width=0.9, height=0.9, name="vp", just=c("right","top"))), action="prepend")
pheatmap(exp.clean, show_rownames = F)
setHook("grid.newpage", NULL, "replace")
grid.text("celllines", y=-0.07, gp=gpar(fontsize=16))
grid.text("genes", x=-0.07, rot=90, gp=gpar(fontsize=16))

setHook("grid.newpage", function() pushViewport(viewport(x=1,y=1,width=0.9, height=0.9, name="vp", just=c("right","top"))), action="prepend")
pheatmap(exp.clean.w0, show_rownames = F)
setHook("grid.newpage", NULL, "replace")
grid.text("celllines", y=-0.07, gp=gpar(fontsize=16))
grid.text("genes", x=-0.07, rot=90, gp=gpar(fontsize=16))
```
Code for other matrices analogous to the one for the expression matrix. 
Observations: for CERES and probability matrix only a few genes seemed to be essential.

# Data reduction
## Additional cleanup beforehand

* Rows containing 0 in matrices were removed, and gene data availability was compared again to ensure all matrices had the same dimension.

```
sum(exp.clean == 0)
exp.clean.w0 <- exp.clean[!(apply(exp.clean, 1, function(y) any(y == 0))),]
sum(exp.clean.w0 == 0)

sum(copy.clean == 0)
copy.clean.w0 <- copy.clean[!(apply(copy.clean, 1, function(y) any(y == 0))),]
sum(copy.clean.w0 == 0)

sum(ceres.clean == 0)
ceres.clean.w0 <- ceres.clean[!(apply(ceres.clean, 1, function(y) any(y == 0))),]
sum(ceres.clean.w0 == 0)

sum(prob.clean == 0)
prob.clean.w0 <- prob.clean[!(apply(prob.clean, 1, function(y) any(y == 0))),]
sum(prob.clean.w0 == 0)

dim(copy.clean.w0) == dim(exp.clean.w0)
gene.data.ex = c(rownames(exp.clean.w0))
copy.clean.w0 = copy.clean.w0[-which(rownames(copy.clean.w0) %!in% gene.data.ex),]
genes.clean.w0 = rownames(exp.clean.w0)
ceres.clean.w0 = ceres.clean.w0[-which(rownames(ceres.clean.w0) %!in% genes.clean.w0),]
prob.clean.w0 = prob.clean.w0[-which(rownames(prob.clean.w0) %!in% genes.clean.w0),]
genes.clean.w0 = rownames(prob.clean.w0)
exp.clean.w0 = exp.clean.w0[-which(rownames(exp.clean.w0) %!in% genes.clean.w0),]
copy.clean.w0 = copy.clean.w0[-which(rownames(copy.clean.w0) %!in% genes.clean.w0),]
ceres.clean.w0 = ceres.clean.w0[-which(rownames(ceres.clean.w0) %!in% genes.clean.w0),]
rm(gene.data.ex)
```
dimensions: 11545x28

# Principal component analysis (PCA)
## A principal component analysis was performed with each matrix (probability, CN, CERES, expression) to find out if there are striking correlations between all the samples.

* First we did a barplot with the proportional variances  
```
pca_prob <- prcomp(t(prob.clean.w0), scale = TRUE) 
plot(pca_prob$x[,1], pca_prob$x[,2]) 
pca_prob_var <- pca_prob$sdev^2 
pca_prob_var_per <- round(pca_prob_var/sum(pca_prob_var)*100, 1) 
barplot(pca_prob_var_per, main = "Proportion of variance", xlab = "Principal component", ylab = "Percent variation")
```

* Then we plotted the principal components (PC'S) against each other.

Plot PC1 against PC2:
```
pca_data_prob_12 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,1], Y = pca_prob$x[,2]) 
pca_data_prob_12
pca_prob_plot_12 <- ggplot(data = pca_data_prob_12, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC1 -", pca_prob_var_per[1], "%", sep = ""))  + ylab(paste("PC2 -", pca_prob_var_per[2], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_12
```

Plot PC1 against PC3:
```
pca_data_prob_13 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,1], Y = pca_prob$x[,3]) 
pca_data_prob_13
pca_prob_plot_13 <- ggplot(data = pca_data_prob_13, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC1 -", pca_prob_var_per[1], "%", sep = ""))  + ylab(paste("PC3 -", pca_prob_var_per[3], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_13
```

Plot PC1 against PC4:
```
pca_data_prob_14 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,1], Y = pca_prob$x[,4]) 
pca_data_prob_14
pca_prob_plot_14 <- ggplot(data = pca_data_prob_14, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC1 -", pca_prob_var_per[1], "%", sep = ""))  + ylab(paste("PC4 -", pca_prob_var_per[4], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_14
```

Plot PC1 against PC5:
```
pca_data_prob_15 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,1], Y = pca_prob$x[,5]) 
pca_data_prob_15
pca_prob_plot_15 <- ggplot(data = pca_data_prob_15, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC1 -", pca_prob_var_per[1], "%", sep = ""))  + ylab(paste("PC5 -", pca_prob_var_per[5], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_15
```

Plot PC2 against PC3:
```
pca_data_prob_23 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,2], Y = pca_prob$x[,3]) 
pca_data_prob_23
pca_prob_plot_23 <- ggplot(data = pca_data_prob_23, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC2 -", pca_prob_var_per[2], "%", sep = ""))  + ylab(paste("PC3 -", pca_prob_var_per[3], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_23
```

Plot PC2 against PC4:
```
pca_data_prob_24 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,2], Y = pca_prob$x[,4]) 
pca_data_prob_24
pca_prob_plot_24 <- ggplot(data = pca_data_prob_24, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC2 -", pca_prob_var_per[2], "%", sep = ""))  + ylab(paste("PC4 -", pca_prob_var_per[4], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_24
````

Plot PC2 against PC5:
```
pca_data_prob_25 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,2], Y = pca_prob$x[,5]) 
pca_data_prob_25
pca_prob_plot_25 <- ggplot(data = pca_data_prob_25, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC2 -", pca_prob_var_per[2], "%", sep = ""))  + ylab(paste("PC5 -", pca_prob_var_per[5], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_25
```

Plot PC3 against PC4:
```
pca_data_prob_34 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,3], Y = pca_prob$x[,4]) 
pca_data_prob_34
pca_prob_plot_34 <- ggplot(data = pca_data_prob_34, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC3 -", pca_prob_var_per[3], "%", sep = ""))  + ylab(paste("PC4 -", pca_prob_var_per[4], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_34
```

Plot PC3 against PC5:
```
pca_data_prob_35 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,3], Y = pca_prob$x[,5]) 
pca_data_prob_35
pca_prob_plot_35 <- ggplot(data = pca_data_prob_35, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC3 -", pca_prob_var_per[3], "%", sep = ""))  + ylab(paste("PC5 -", pca_prob_var_per[5], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_35
```

Plot PC4 against PC5:
```
pca_data_prob_45 <- data.frame(Sample = rownames(pca_prob$x), X = pca_prob$x[,4], Y = pca_prob$x[,5]) 
pca_data_prob_45
pca_prob_plot_45 <- ggplot(data = pca_data_prob_45, aes(x=X, y=Y, label = Sample)) + geom_point()   + xlab(paste("PC4 -", pca_prob_var_per[4], "%", sep = ""))  + ylab(paste("PC5 -", pca_prob_var_per[5], "%", sep="")) + theme_bw() + ggtitle("PCA probability score")
pca_prob_plot_45
```
* After plotting the PC's we determined the genes with higher variance 

Loading scores PC1:
```
loading_scores_prob_1 <- pca_prob$rotation[,1]      
gene_scores_prob_1 <- abs(loading_scores_prob_1)          
gene_score_prob_ranked_1 <- sort(gene_scores_prob_1, decreasing = TRUE)
top_10_genes_prob_1 <- names(gene_score_prob_ranked_1[1:10])
top_10_genes_prob_1
pca_prob$rotation[top_10_genes_prob_1,1]
```

Loading scores P2:
```
loading_scores_prob_2 <- pca_prob$rotation[,2]      
gene_scores_prob_2 <- abs(loading_scores_prob_2)          
gene_score_prob_ranked_2 <- sort(gene_scores_prob_2, decreasing = TRUE)
top_10_genes_prob_2 <- names(gene_score_prob_ranked_2[1:10])
top_10_genes_prob_2
pca_prob$rotation[top_10_genes_prob_2,2]
```

Loading scores PC3:
```
loading_scores_prob_3 <- pca_prob$rotation[,3]      
gene_scores_prob_3 <- abs(loading_scores_prob_3)          
gene_score_prob_ranked_3 <- sort(gene_scores_prob_3, decreasing = TRUE)
top_10_genes_prob_3 <- names(gene_score_prob_ranked_3[1:10])
top_10_genes_prob_3
pca_prob$rotation[top_10_genes_prob_3,3]
```

Loading scores PC4:
```
loading_scores_prob_4 <- pca_prob$rotation[,4]      
gene_scores_prob_4 <- abs(loading_scores_prob_4)          
gene_score_prob_ranked_4 <- sort(gene_scores_prob_4, decreasing = TRUE)
top_10_genes_prob_4 <- names(gene_score_prob_ranked_4[1:10])
top_10_genes_prob_4
pca_prob$rotation[top_10_genes_prob_4,4]
```

Loading scores PC5:
```
loading_scores_prob_5 <- pca_prob$rotation[,5]      
gene_scores_prob_5 <- abs(loading_scores_prob_5)          
gene_score_prob_ranked_5 <- sort(gene_scores_prob_5, decreasing = TRUE)
top_10_genes_prob_5 <- names(gene_score_prob_ranked_5[1:10])
top_10_genes_prob_5
pca_prob$rotation[top_10_genes_prob_5,5]
```

* PCA's for other matrices done analogous to the probability matrix.



